src/core/mac/mac.cpp

```diff
@@ -435,7 +438,10 @@ Error Mac::SetPanChannel(uint8_t aChannel)
     mRadioChannel = mPanChannel;
 
 #if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
-    UpdateCsl();
+    if ((mCslChannel == 0) && isRadioChannelChanged)
+    {
+        UpdateCslParameters();
+    }
```

Comment:
Should we check and use `mPanChannel` and not `mRadioChannel`?

I see that we previously used `mRadioChannel`, but I think it was a mistake. `mPanChannel` is the channel the Thread PAN is operating on. `mRadioChannel` is the current channel the radio is listening on, which can vary (e.g., during a discovery scan, we may switch and listen to a bunch of channels, which would change `mRadioChannel` but keep `mPanChannel` unchanged).

Maybe would be better if we fix this in a small seperate PR?

Reply:
Submitted https://github.com/openthread/openthread/pull/11336